app/services/cosmos/cosmos.ts

```diff
@@ -1,63 +1,121 @@
-import Cosmos from "@lunie/cosmos-api"
+/* eslint-disable @typescript-eslint/camelcase */
+import {
+  DistributionExtension,
+  QueryClient,
+  setupDistributionExtension,
+  setupStakingExtension,
+  StargateClient,
+  StakingExtension,
+  SigningStargateClient,
+  BroadcastTxResponse,
+  MsgSendEncodeObject,
+  MsgDelegateEncodeObject,
+  MsgUndelegateEncodeObject,
+  MsgWithdrawDelegatorRewardEncodeObject,
+} from "@cosmjs/stargate";
+import { OfflineDirectSigner } from '@cosmjs/proto-signing';
+import { Tendermint34Client } from "@cosmjs/tendermint-rpc";
+import { TextDecoder } from 'text-decoding';
+import BigNumber from "bignumber.js";
 
+import { MsgSend } from 'cosmjs-types/cosmos/bank/v1beta1/tx';
+import {
+  MsgBeginRedelegate,
+  MsgDelegate,
+  MsgUndelegate,
+} from 'cosmjs-types/cosmos/staking/v1beta1/tx'
+import { MsgWithdrawDelegatorReward } from "cosmjs-types/cosmos/distribution/v1beta1/tx";
 import {
-  CosmosAccountResult,
   CosmosCoinResult,
   CosmosDelegation,
   CosmosMessage,
+  CosmosMessageToSign,
+  CosmosRedelegation,
   CosmosRewardsResult,
+  CosmosSigner,
   CosmosUnbondingDelegation,
   CosmosValidator,
 } from "./cosmos.types"
 import {
-  extractCoinFromCosmosCoinList,
-  parseCosmosCoin,
+  convertValidator,
+  convertDecCoin,
+  convertDelegationDelegatorReward,
+  convertDelegationResponse,
+  convertUnbondingDelegation,
+  convertRedelegation,
 } from "./cosmos.utils"
+import { MintExtension, setupMintExtension } from "./mint-query-extension"
 
 /**
  * Cosmos API helper for LikeCoin
  */
 export class CosmosAPI {
-  /**
-   * The Cosmos API client
-   */
-  api: Cosmos
+  defaultGasLimits = {
+    send: 80000,
+    delegate: 160000,
+    transfer: 160000,
+    undelegate: 160000,
+    withdraw: 160000,
+  }
+
+  restURL: string
+
+  stargateClient: StargateClient
+
+  queryClient: QueryClient & DistributionExtension & StakingExtension & MintExtension
+
+  async setup(restURL: string) {
+    this.restURL = restURL
+    this.stargateClient = await StargateClient.connect(restURL);
 
-  setup(restURL: string, chainId: string) {
-    this.api = new Cosmos(restURL, chainId)
+    const tendermint34Client = await Tendermint34Client.connect(restURL);
+    this.queryClient = QueryClient.withExtensions(
+      tendermint34Client,
+      setupDistributionExtension,
+      setupStakingExtension,
+      setupMintExtension,
+    )
   }
 
   /**
    * Get the list of validators
    */
-  async getValidators() {
-    return this.api.get.validators() as CosmosValidator[]
+  async getValidators(): Promise<CosmosValidator[]> {
+    const bondStatus = 'BOND_STATUS_BONDED'
```

Comment:
What is the meaning of `BOND_STATUS_BONDED` ?

Reply:
It comes from [here](https://github.com/confio/cosmjs-types/blob/main/src/cosmos/staking/v1beta1/staking.ts#L13).
I also saw the [testcase](https://github.com/cosmos/cosmjs/blob/main/packages/stargate/src/queries/staking.spec.ts#L238) uses the `BOND_STATUS_BONDED` to query validators.
Should we use another status?

Reply:
I think bond means active, should be ok